keps/sig-storage/5489-default-storage-class-per-access-mode/README.md

```diff
@@ -0,0 +1,908 @@
+
+This limitation has been a long-standing issue within the community, as evidenced by the extensive discussion in [Kubernetes Issue #89911](https://github.com/kubernetes/kubernetes/issues/89911), which was first opened in 2020. The sustained engagement on this issue demonstrates a clear and persistent demand from users and cluster administrators for a more flexible and intelligent defaulting mechanism.
+
+### Goals
+
+* To allow cluster administrators to specify a default `StorageClass` for each distinct volume access mode (`ReadWriteOnce`, `ReadWriteMany`, `ReadOnlyMany`, etc.).
+* To improve the user experience by automatically provisioning volumes from the correct storage backend based on the access mode requested in a `PersistentVolumeClaim` (PVC), without requiring the user to specify a `storageClassName`.
+* To reduce provisioning errors in clusters with multiple types of storage systems.
+* To maintain backward compatibility with the existing single-default `StorageClass` mechanism, which will serve as a fallback.
+* To define a deterministic defaulting behavior for PVCs that list multiple access modes. The system will use a fixed priority order (`ReadWriteMany` > `ReadOnlyMany` > `ReadWriteOnce` > `ReadWriteOncePod`) to select a default `StorageClass` if the PVC requests multiple modes for which defaults exist.
+
+### Non-Goals
+
+* This KEP will not introduce new `API` objects. The proposed solution will be implemented using annotations on the existing `StorageClass` object.
+* This KEP will not change the behavior for `PersistentVolumeClaims` (PVCs) that explicitly set a `storageClassName`.
+* This KEP will not alter the validation logic that ensures a provisioner supports the access modes it is being asked to provision for.
+
+## Proposal
+
+This KEP proposes enhancing the `PersistentVolumeClaim` (PVC) admission process to support defining a default `StorageClass` based on the volume access mode. We will introduce a new annotation that can be applied to `StorageClass` objects to designate them as the default for a specific access mode.
+
+**This design will not impact the current quota calculation.** The mechanism for resource quota enforcement on PVCs remains unchanged; only the defaulting logic for `storageClassName` is affected.
+
+The proposed annotation is `storageclass.kubernetes.io/is-default-class-for-mode`. A cluster administrator can apply this annotation to multiple `StorageClass` objects, each with a different access mode value, such as `"ReadWriteOnce"`, `"ReadWriteMany"`, `"ReadOnlyMany"`, or `"ReadWriteOncePod"`.
```

Comment:
would it make sense to call `storageclass.kubernetes.io/is-default-class-for-mode` as `storageclass.kubernetes.io/is-default-class-for-access-mode` so that we can extend for volume mode something like `storageclass.kubernetes.io/is-default-class-for-volume-mode` in future?